### ISSUE TITLE
MD039 Spaces inside link text

### ISSUE THREAD

**Issue by peter1000** [markdownlint/markdownlint#102]
This rule is triggered on links that have spaces surrounding the link text

It also triggers when I use an emphasis within a link text.

Example:

``` markdown
[The **test** web site](http://www.example.com/)
```

Not sure if that is a bug but I wanted to report it.

test.md:2: MD039 Spaces inside link text

**Comment by shichao-an** [comment]
Same to backticks:

```
[is of `uint8` type](http://www.example.com/)
```

This will also trigger MD039.

**Comment by astorije** [comment]
I'm seeing this issue with backticks as well. Specifically:

``` md
[open a PR for `ipsum`](https://github.com/lorem/ipsum/compare/my-branch?expand=1)
```

**Comment by gregswift** [comment]
Inline HTML will also trigger MD039

```
[Common <br> templates][1]
```

Found this from formatting inside of a table.
